### PR TITLE
Fix pac config not getting added

### PR DESCRIPTION
**Pull request by piyush-garg**
This will fix the issue that pac config is not getting added in spec and resulting in config always false

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Run `make test lint` before submitting a PR
- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/operator/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

```release-note
Fix pac config not getting added
```

### CODE REVIEW

File: pkg/apis/operator/v1alpha1/openshiftpipelinesascode_defaults.go [hunk 1]
```diff
@@ -65,3 +69,39 @@ func setAdditionalPACControllerDefault(additionalPACController map[string]Additi
...
+	structType := reflect.TypeOf(settings).Elem()
+	config := map[string]string{}
+
+	for i := 0; i < structType.NumField(); i++ {
+		field := structType.Field(i)
+		fieldName := field.Name
+
+		jsonTag := field.Tag.Get("json")
+		// Skip field which doesn't have json tag
+		if jsonTag == "" {
```
[the remark this answers is not in the document]
Comment (jkandasa) [reply pkg/apis/operator/v1alpha1/openshiftpipelinesascode_defaults.go:84]: also we may get `omitempty` on the json tag

File: pkg/apis/operator/v1alpha1/openshiftpipelinesascode_defaults.go [hunk 2]
```diff
@@ -65,3 +69,39 @@ func setAdditionalPACControllerDefault(additionalPACController map[string]Additi
...
+		}
+
+		//nolint
+		switch field.Type.Kind() {
+		case reflect.String:
+			config[strings.ToLower(jsonTag)] = structValue.FieldByName(fieldName).String()
+		case reflect.Bool:
+			config[strings.ToLower(jsonTag)] = strconv.FormatBool(structValue.FieldByName(fieldName).Bool())
+		case reflect.Int:
+			if structValue.FieldByName(fieldName).Int() == 0 {
```
[the remark this answers is not in the document]
Comment (piyush-garg) [reply pkg/apis/operator/v1alpha1/openshiftpipelinesascode_defaults.go:95]: i have made the reverse here of this https://github.com/openshift-pipelines/pipelines-as-code/blob/main/pkg/configutil/config.go#L78

we should add the field in configmap with default ""

File: pkg/apis/operator/v1alpha1/openshiftpipelinesascode_defaults.go [hunk 3]
```diff
@@ -65,3 +69,39 @@ func setAdditionalPACControllerDefault(additionalPACController map[string]Additi
...
+			continue
+		}
+
+		//nolint
+		switch field.Type.Kind() {
+		case reflect.String:
+			config[strings.ToLower(jsonTag)] = structValue.FieldByName(fieldName).String()
+		case reflect.Bool:
+			config[strings.ToLower(jsonTag)] = strconv.FormatBool(structValue.FieldByName(fieldName).Bool())
+		case reflect.Int:
```
[the remark this answers is not in the document]
Comment (piyush-garg) [reply pkg/apis/operator/v1alpha1/openshiftpipelinesascode_defaults.go:94]: right now all fields are int only, so not implemented other https://github.com/openshift-pipelines/pipelines-as-code/blob/main/pkg/params/settings/config.go#L43

File: pkg/apis/operator/v1alpha1/openshiftpipelinesascode_defaults.go [hunk 4]
```diff
@@ -65,3 +69,39 @@ func setAdditionalPACControllerDefault(additionalPACController map[string]Additi
...
+	for i := 0; i < structType.NumField(); i++ {
+		field := structType.Field(i)
+		fieldName := field.Name
+
+		jsonTag := field.Tag.Get("json")
+		// Skip field which doesn't have json tag
+		if jsonTag == "" {
+			continue
+		}
+
```
[the remark this answers is not in the document]
Comment (piyush-garg) [reply pkg/apis/operator/v1alpha1/openshiftpipelinesascode_defaults.go:87]: done

File: pkg/apis/operator/v1alpha1/openshiftpipelinesascode_defaults.go [hunk 5]
```diff
@@ -65,3 +69,39 @@ func setAdditionalPACControllerDefault(additionalPACController map[string]Additi
...
+		//nolint
+		switch field.Type.Kind() {
```
[the remark this answers is not in the document]
Comment (piyush-garg) [reply pkg/apis/operator/v1alpha1/openshiftpipelinesascode_defaults.go:89]: done

File: pkg/apis/operator/v1alpha1/openshiftpipelinesascode_defaults.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (piyush-garg) [reply pkg/apis/operator/v1alpha1/openshiftpipelinesascode_defaults.go:84]: i have added `-` and there is no omit empty here so skipped https://github.com/openshift-pipelines/pipelines-as-code/blob/main/pkg/params/settings/config.go#L43

**Review by jkandasa — APPROVED** [review]
/lgtm

**Pull request merged**